### ISSUE TITLE
IllegalArgumentException when supplying an empty list for a schedule

### ISSUE THREAD

**Issue by aterweele** [jarohen/chime#20]
Sometimes, I would like to specify an empty schedule, but when I supply `[]` or `nil` as the schedule arguments to `chime-ch` or `chime-at`, both will attempt to call `clj-time.core/before?` on `nil`:
```
user> (chime/chime-ch nil)
Exception in thread "async-dispatch-1" 
java.lang.IllegalArgumentException: No implementation of method: :before? of protocol: #'clj-time.core/DateTimeProtocol found for class: nil
	at clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:568)
	at clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:560)
	at clj_time.core$eval11045$fn__11090$G__11022__11097.invoke(core.clj:102)
	at chime$ms_between.invokeStatic(chime.clj:8)
	at chime$ms_between.invoke(chime.clj:7)
	at chime$chime_ch$fn__19258$state_machine__17295__auto____19259$fn__19261.invoke(chime.clj:38)
	at chime$chime_ch$fn__19258$state_machine__17295__auto____19259.invoke(chime.clj:38)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:1011)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:1010)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:1015)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:1013)
	at chime$chime_ch$fn__19258.invoke(chime.clj:38)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
#object[chime$chime_ch$reify__19297 0x6f223434 "chime$chime_ch$reify__19297@6f223434"]
```

```
user> (chime/chime-at nil (fn [_] (println "called")))
Exception in thread "async-dispatch-3" 
java.lang.IllegalArgumentException: No implementation of method: :before? of protocol: #'clj-time.core/DateTimeProtocol found for class: nil
	at clojure.core$_cache_protocol_fn.invokeStatic(core_deftype.clj:568)
	at clojure.core$_cache_protocol_fn.invoke(core_deftype.clj:560)
	at clj_time.core$eval11045$fn__11090$G__11022__11097.invoke(core.clj:102)
	at chime$ms_between.invokeStatic(chime.clj:8)
	at chime$ms_between.invoke(chime.clj:7)
	at chime$chime_ch$fn__19258$state_machine__17295__auto____19259$fn__19261.invoke(chime.clj:38)
	at chime$chime_ch$fn__19258$state_machine__17295__auto____19259.invoke(chime.clj:38)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invokeStatic(ioc_macros.clj:1011)
	at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:1010)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invokeStatic(ioc_macros.clj:1015)
	at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:1013)
	at chime$chime_ch$fn__19258.invoke(chime.clj:38)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
#function[chime/chime-at/cancel!--19353]
```
Both calls will return the expected channel/cancel function, but the exception is still worrisome and is causing my tests to fail.

I tried running the likely culprit:

```
    (go-loop [now (t/now)
              [next-time & more-times] (->> (times-fn)
                                            (map tc/to-date-time)
                                            (drop-while #(t/before? % now)))]
      ...)
```

But I could not reproduce the problem in my REPL.

By the way, thanks for a great project! I really like the abstraction provided by `chime-ch`.

**Comment by aterweele** [comment]
On further investigation, I think this issue affects all schedules in which all the times have passed, so the schedule `(map #(-> % clj-time.core/minutes clj-time.core/ago) [5 4 3 2])` will also trigger the issue.

**Comment by jarohen** [comment]
Hi Alex, thanks for the bug report! Should just require a quick check for an empty seq in that `go-loop` - will take a look now

**Comment by jarohen** [comment]
Have released 0.2.1 with a bugfix - it now explicitly checks for an empty sequence of times. Let me know if this fixes your issue :)

James

**Comment by aterweele** [comment]
Yup, all good! Thanks again 🙂
